app/models/pipeline_run.rb

```diff
@@ -542,16 +542,31 @@ def db_load_contigs(contig2taxid)
     taxon_lineage_map = {}
     TaxonLineage.where(taxid: taxid_list.uniq).order(:id).each { |t| taxon_lineage_map[t.taxid.to_i] = t.to_a }
 
+    # A lambda allows us to access variables in the enclosing scope, such as contig2taxid.
+    get_contig_hash = lambda do |header, sequence|
+      read_count = contig_stats_json[header] || 0
+      lineage_json = get_lineage_json(contig2taxid[header], taxon_lineage_map)
+      species_taxid_nt = lineage_json.dig("NT", 0) || TaxonLineage::MISSING_SPECIES_ID
```

Comment:
As discussed, lets not use MISSING ids as it can lead to errors when merging with taxon counts.